demos/classification_benchmark_demo/cpp/main.cpp

```diff
@@ -118,229 +127,220 @@ cv::Mat centerSquareCrop(const cv::Mat& image) {
-                } else {
-                    imageNames[i] = name;
-                }
+        std::string line;
+        while (std::getline(inputGtFile, line)) {
+            size_t separatorIdx = line.find(' ');
+            if (separatorIdx == std::string::npos) {
+                throw std::runtime_error("The ground truth file has incorrect format.");
+            }
+            std::string imagePath = line.substr(0, separatorIdx);
+            size_t imagePathEndIdx = imagePath.rfind('/');
+            unsigned classIndex = static_cast<unsigned>(std::stoul(line.substr(separatorIdx + 1)));
+            if ((imagePathEndIdx != 1 || imagePath[0] != '.') && imagePathEndIdx != std::string::npos) {
+                throw std::runtime_error("The ground truth file has incorrect format.");
             }
+            classIndicesMap.insert({imagePath.substr(imagePathEndIdx + 1), classIndex});
         }
 
-        // ----------------------------------------Read image classes-----------------------------------------
-        std::vector<unsigned> classIndices;
-        if (!FLAGS_gt.empty()) {
-            std::map<std::string, unsigned> classIndicesMap;
-            std::ifstream inputGtFile(FLAGS_gt);
-            if (!inputGtFile.is_open()) {
-                throw std::runtime_error("Can't open the ground truth file.");
+        for (size_t i = 0; i < imageNames.size(); i++) {
+            auto imageSearchResult = classIndicesMap.find(imageNames[i]);
+            if (imageSearchResult != classIndicesMap.end()) {
+                classIndices.push_back(imageSearchResult->second);
+            } else {
+                throw std::runtime_error("No class specified for image " + imageNames[i]);
             }
+        }
+    } else {
+        classIndices.resize(inputImages.size());
+        std::fill(classIndices.begin(), classIndices.end(), 0);
+    }
 
-            std::string line;
-            while (std::getline(inputGtFile, line)) {
-                size_t separatorIdx = line.find(' ');
-                if (separatorIdx == std::string::npos) {
-                    throw std::runtime_error("The ground truth file has incorrect format.");
-                }
-                std::string imagePath = line.substr(0, separatorIdx);
-                size_t imagePathEndIdx = imagePath.rfind('/');
-                unsigned classIndex = static_cast<unsigned>(std::stoul(line.substr(separatorIdx + 1)));
-                if ((imagePathEndIdx != 1 || imagePath[0] != '.') && imagePathEndIdx != std::string::npos) {
-                    throw std::runtime_error("The ground truth file has incorrect format.");
-                }
-                classIndicesMap.insert({imagePath.substr(imagePathEndIdx + 1), classIndex});
+    //------------------------------ Running routines ----------------------------------------------
+    std::vector<std::string> labels = ClassificationModel::loadLabels(FLAGS_labels);
```

Comment:
`--labels` is now optional. Handle if `FLAGS_labels` is empty.

Mark `--labels <LABELS>` with [] in help